### PR TITLE
ensure all popovers have aria-describedBy

### PR DESCRIPTION
**Pull request by peetucket**
# Why was this change made? 🤔

Fixes one accessibility item in #3171 

Note, after looking at https://inclusive-components.design/tooltips-toggletips/#inclusivetoggletips will leave this as aria-describedby tags, but have now moved the element they are attached to, in some cases to the form input.  In some cases, the tip is describing a section, so left on the label or legend in those cases.

# How was this change tested? 🤨

Localhost

### CODE REVIEW

**Review by justinlittman — CHANGES_REQUESTED** [review]
Also, can you add comments to PopoverComponent describing the accessibility-correct way to use it.

File: app/components/first_draft_collections/create/details_component.html.erb [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <header>Details *</header>
   <div class="mb-3 row">
     <div class="col-sm-2">
-      <%= form.label :name, "Collection name *", class: "col-form-label" %>
+      <%= form.label :name, "Collection name *", class: "col-form-label", "aria-describedby": "popover-collection.name" %>
```
[the remark this answers is not in the document]
Comment (peetucket) [reply app/components/first_draft_collections/create/details_component.html.erb:5]: good question - i was assuming the description was of the input, but i could be convinced it was the label too... we can verify with an expert

File: app/components/collections/edit_license_component.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-  <section id="terms" data-controller="complex-radio">
+  <section id="terms" data-controller="complex-radio" aria-describedby="popover-collection.terms">
```
[the remark this answers is not in the document]
Comment (peetucket) [reply app/components/collections/edit_license_component.html.erb:1]: oops...this change adding the corresponding`id` is on the `PopoverComponent` and incorrectly got pulled into a different already merged PR.  The ID is now added to any element via the `PopoverComponent` in main: https://github.com/sul-dlss/happy-heron/blob/main/app/components/popover_component.html.erb#L1

The ID is based on the key, which is the change I added here.

File: app/components/first_draft_collections/create/details_component.html.erb (hunk 1 above)
[the remark this answers is not in the document]
Comment (peetucket) [reply app/components/first_draft_collections/create/details_component.html.erb:5]: I think this is correct ... it is and should be the label, this is what is the tooltip is right next to and is what it is tied to

File: app/components/collections/edit_license_component.html.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (peetucket) [reply app/components/collections/edit_license_component.html.erb:1]: I rebased from main to ensure the needed change from `PopoverComponent` is visible in this PR.

**Comment by peetucket** [comment]
> Also, can you add comments to PopoverComponent describing the accessibility-correct way to use it.

I added a comment to the top of `PopoverComponent` explaining how to use it.

File: app/components/first_draft_collections/create/details_component.html.erb (hunk 1 above)
[the remark this answers is not in the document]
Comment (justinlittman) [reply app/components/first_draft_collections/create/details_component.html.erb:5]: Right, but the label is tied to the input. So if the label is describing the input then isn't the tooltip also describing the input?

**Review by justinlittman — APPROVED** [review]
Approve and defer to you on determining whether popover should reference the label or input.

**Comment by peetucket** [comment]
Note, after looking at https://inclusive-components.design/tooltips-toggletips/#inclusivetoggletips will leave this as is with aria-describedby tags, but have now moved the element they are attached to in some cases.  Where it made sense, I moved to the form input being described. In some cases the tip is describing a section, so left on the label or legend in those cases.

**Pull request merged**